software/firmware/europi.py

```diff
@@ -59,6 +59,18 @@
 CHAR_HEIGHT = 8
 
 
+
+# EuroPi error class
+
+class EuroPiError(Exception):
+    def __init__(self, error_code):
+        oled.fill(0)
+        oled.centre_text(f'Error {error_code}')
+        oled.show()
+        time.sleep(5)
+        oled.fill(0)
+
+

```

Comment:
[discussion] We can have standard Exceptions like the following:
```suggestion
class EuroPiValueError(EuroPiError):
    ERROR_CODE = 1
    def __init__(self, message="Unexpected value received"):
        super(message)
````
This associates standard error codes directly with a raisable Exception and provides a default error message that can be overridden at raise time.

For example:
```python
    def set_samples(self, samples):
        """Override the default number of sample reads with the given value."""
        if not isinstance(samples, int):
            raise EuroPiValueError(
                f"set_samples expects an int value, got: {samples}")
        self._samples = samples
```

With the above example, the oled will display "Error: 1" and the terminal stderr will show the error message.